### PR TITLE
Extract components list function for Altium from BOM

### LINKED ISSUE #118: Refactor Altium BOM generation to use list_components
Like OrCAD, we should have a `list_components` intermediate step for generating a list of components for Altium as well.

### PR DESCRIPTION
**Pull request by shrik450**
Closes #118.

Extracts a `list_components_for_altium` function from the BOM generation utility, and uses it for BOM generation as well. This is easier to do now as there are more changes incoming for components listing such as combining multi-symbol components.

Stacked on top of #119 as that should get merged before this.

### CODE REVIEW

**Review by jtran — APPROVED** [review]

**Comment by shrik450** [comment]
I'm holding off on a rebase until #119 is merged to avoid having to re-do it with every change.

**Review by kdumontnu — APPROVED** [review]
👍 sorry, didn’t realize this was blocking the chain of PRs

**Pull request merged**
